ts/date-formatter.ts

```diff
@@ -0,0 +1,82 @@
+import { getWeekDates } from "./dateManipulation.js";
+
+export type DateFormatter = ReturnType<typeof createDateFormatter>;
+
+export const createDateFormatter = (locale: string) => {
```

Comment:
You've defined a locale type right ?
Use it, you can thank me for the bugfix later 😄